File: python/algorithms/quick_sort.py
def quick_sort(arr, start, end):

    if len(arr) == 1:
        return arr

    if start < end:
        p_index = partition(arr, start, end)

        quick_sort(arr, start, p_index-1)
        quick_sort(arr, p_index+1, end)


def partition(arr, start, end):
    p_index = start
    pivot = arr[end]

    for i in range(start, end+1):
        if arr[i] < pivot:
            arr[i], arr[p_index] = arr[p_index], arr[i]
            p_index += 1

    arr[p_index], arr[end] = arr[end], arr[p_index]
    return p_index


if __name__ == "__main__":
    nums = [7, 2, 1, 6, 8, 5, 3, 4]
    quick_sort(nums, 0, len(nums)-1)
    print(nums)

File: c#/algorithms/quickSort.cs
using System;

public class Program
{
  public static void Main()
  {
    int[] nums = new int[] { 7, 2, 1, 6, 8, 5, 3, 4 };
    quickSort(nums, 0, nums.Length - 1);
    foreach (int num in nums)
    {
      Console.WriteLine(num.ToString());
    }
  }

  public static void quickSort(int[] arr, int start, int end)
  {
    if (start >= end)
    {
      return;
    }
    else
    {
      int pIndex = partition(arr, start, end);
      quickSort(arr, start, pIndex - 1);
      quickSort(arr, pIndex + 1, end);
    }
  }

  public static int partition(int[] arr, int start, int end)
  {
    int pIndex = start;
    int pivot = arr[end];
    for (int i = start; i <= end; i++)
    {
      if (arr[i] < pivot)
      {
        int temp = arr[i];
        arr[i] = arr[pIndex];
        arr[pIndex] = temp;
        pIndex++;
      }
    }
    int swap = arr[pIndex];
    arr[pIndex] = arr[end];
    arr[end] = swap;
    return pIndex;
  }
}